### PR TITLE
fix: assets upload to use the correct variable name.

### PR DESCRIPTION
**Pull request by yash-zededa**
Added needs on for the upload job to get the upload url from the create release job.

### CODE REVIEW

File: .github/workflows/assets.yml [hunk 1]
```diff
@@ -117,7 +119,7 @@ jobs:
             # Add ARCH prefix
             new_name="${ARCH}.${base_name}"
             # Rename the file
-            mv "$asset" "assets/$new_name"
+            mv "$file" "assets/$new_name"
```
Comment (OhmSpectator) [inline .github/workflows/assets.yml:122]: Is the same should be done here?
https://github.com/lf-edge/eve/blob/d38d00782bc4e151704771e6f466e1673996ac39/.github/workflows/assets.yml#L118

Comment (yash-zededa) [reply .github/workflows/assets.yml:122]: Yes, fixed!

**Review by uncleDecart — APPROVED** [review]
LGTM

File: .github/workflows/assets.yml [hunk 2]
```diff
@@ -113,11 +115,11 @@ jobs:
           # Create SHA256 checksum for rootfs.img
           sha256sum "assets/rootfs.img" | awk '{ print $1 }' > "assets/rootfs.img.sha256"
```
Comment (eriknordmark) [inline .github/workflows/assets.yml:116]: I thought the sha256 failed? Did I misread the error?

Comment (yash-zededa) [reply .github/workflows/assets.yml:116]: The failure was not related to the sha. The mv command was failing as the variable name was incorrect.

Comment (yash-zededa) [reply .github/workflows/assets.yml:116]: @eriknordmark iIf the changes look good, I'd like to merge this, backport it to 13.6, and create a tag to test the workflow.

**Review by deitch — APPROVED** [review]

**Pull request merged**